crates/ra_ide_api/src/display/navigation_target.rs

```diff
@@ -290,83 +407,54 @@ impl NavigationTarget {
             focus_range,
             // ptr: Some(LocalSyntaxPtr::new(node)),
             container_name: None,
+            description,
+            docs,
         }
     }
+}
 
-    pub(crate) fn node(&self, db: &RootDatabase) -> Option<TreeArc<SyntaxNode>> {
-        let source_file = db.parse(self.file_id()).tree;
-        let source_file = source_file.syntax();
-        let node = source_file
-            .descendants()
-            .find(|node| node.kind() == self.kind() && node.range() == self.full_range())?
-            .to_owned();
-        Some(node)
-    }
+fn docs_from_symbol(db: &RootDatabase, symbol: &FileSymbol) -> Option<String> {
```

Comment:
Should this function and `description_from_symbol` move to `FileSymbol` ? Or should it be a field in `FileSymbol` ? I

Reply:
Good question. I think makeing these fields of symbols is best

Reply:
Otoh, I think current solution is also great! Long term, we probably should migrate symbols to hir as well